### PR TITLE
#311: List more than one example on the Examples page :)

### PR DESCRIPTION
**Pull request by webchick**
(Corresponding Jira ticket EDU-1842)

## What was changed

This PR adds a few more "non-controversial" examples (meaning, limited to the github.com/temporalio namespace for now) to the [Examples](https://learn.temporal.io/examples/) page, including a reference to the new [Order Management System](https://github.com/temporalio/reference-app-orders-go) which was just launched at Replay, so we can give people a bit more of a head start.

It also introduces the concept of "categories" on this page, so we can expand the list here to other types of examples in the future.

## Why?
See #311 for background/rationale.

## Checklist

1. Pushes forward #311 (maybe doesn't "close" it)

2. How was this tested:
I ran `vale index.md` as instructed in the README and proactively fixed a few things that it found. (It unfortunately barfs on the link titles to the SDK examples though :\)

I couldn't seem to get the link checking working (I blame midnight oil :D) so I did a manual click-through on those.

3. Any docs updates needed?
N/A?

### CODE REVIEW

**Comment by webchick** [comment]
A bikeshed we could reasonably paint here btw, is what categories make the most sense for the _people perusing this content_ (vs. how "we" think of them); however, I once again underscore that right now the examples page has only ONE example, so I would prefer to defer that painting party to another time. ;)

**Comment by axfelix** [comment]
These are nice additions and the build is passing which is rare for first-time learn or docs site contributors, so thank you very much for that!

I think we've been meaning to have a longer conversation for a while about whether we want to link to the sample repos directly or whether we want to try to progressively write them up as tutorials, so this might be an open-ended addition -- and likewise for the OMS, I don't know how we're planning to socialize that at the moment.

So basically -- this PR should absolutely stay open for now, but I'm not 100% on where we stand on merging it currently.

**Comment by axfelix** [comment]
@temporalio/education

**Comment by webchick** [comment]
> I think we've been meaning to have a longer conversation for a while about whether we want to link to the sample repos directly or whether we want to try to progressively write them up as tutorials

Makes sense. Here are a few data points to consider:

1. Right now, it takes a person on average between 4-6 months to go between "I just learned about Temporal" to "I built my first POC using it." That is an _awfully_ long time to get over the [suck threshold](https://headrush.typepad.com/creating_passionate_users/2005/10/getting_users_p.html).
2. During both discussions internally (with PMM, SAs), as well as externally (with customers + users at Replay), a lack of examples to start from was _widely_ cited as the top reason this tends to happen.
3. If you do a Google search for "temporal examples," https://learn.temporal.io/examples/ is the top result (nicely done!) so providing people with a strong first impression here feels quite important.

If we link directly to repos as a starting point, we can easily change those links over to tutorials as they become available. But in the meantime, all 3 of the above points are improved.

**Comment by axfelix** [comment]
@napcs  For your consideration next week while I'm out

**Review by napcs — APPROVED** [review]

**Comment by napcs** [comment]
This is the right first step and I appreciate @webchick for taking the time to make this happen.
